certification/internal/policy/operator/deployable_by_olm.go

```diff
@@ -157,19 +158,28 @@ func (p *DeployableByOlmCheck) isCSVReady(installedCSV string, operatorData Oper
 			if csv.Status.Phase == operatorv1alpha1.CSVPhaseSucceeded {
 				log.Debug("CSV is created successfully: ", installedCSV)
 				csvReadyDone <- fmt.Sprintf("%#v", csv)
+				return nil
 			}
 			log.Debug("CSV is not ready yet, retrying...")
-			time.Sleep(2 * time.Second)
+
+			select {
+			case <-ctx.Done():
+				log.Error(fmt.Sprintf("failed to fetch the csv %s: ", installedCSV), ctx.Err())
+				contextTimeOut <- ctx.Err()
+				return ctx.Err()
+			default:
+				time.Sleep(2 * time.Second)
+			}
 		}
-	}()
+	}(ctx)
 
 	select {
 	case csv := <-csvReadyDone:
 		return len(csv) > 0, nil
-	case <-ctx.Done():
```

Comment:
I don't think you need the extra contextTimeOut channel. This should be sufficient.